core/src/main/scala/com/salesforce/op/stages/impl/tuning/Splitter.scala

```diff
@@ -74,7 +67,17 @@ abstract class Splitter(val uid: String) extends SplitterParams {
    * @param data
    * @return Training set test set
    */
-  def prepare(data: Dataset[Row]): ModelData
+  def prepare(data: Dataset[Row]): Dataset[Row]
+
+
+  /**
+   * Function to use examine the data set to set parameters for preparation
+   * eg - do data balancing or dropping based on the labels
+   *
+   * @param data
+   * @return Parameters set in examining data
+   */
+  def examine(data: Dataset[Row]): Option[SplitterSummary]
```

Comment:
this function is designed to have a side-effects for prepare whereas examine sounds read-only. Can we call it something like `setupPrepare` or `prePrepare`?

Reply:
what about assesForPrepare?

Reply:
same read-only connotation, I thin your current is `preValidationPrepare` is good